PR: https://github.com/odoo/odoo/pull/57014

From: 8b5603bad0c6dbe1abc2919ec0670677841c164f
From: Arnaud Joset
Date: 2020-09-14 08:27:44

Breaking data model changes score: 3, change matches:
-    _inherit = 'res.company'
-    pad_server = fields.Char(related='company_id.pad_server', string="Pad Server", readonly=False)
-    pad_key = fields.Char(related='company_id.pad_key', string="Pad API Key", readonly=False)

Total Changes: 38

[IMP] pad: save pas configuration in ir.config_parameter

Before this commit, the pad key and server were saved on the company. It was an historical decision whose reasons no longer applies.
The "Root url" called later "URL template" was company dependant but it no longer exists.

closes odoo/odoo#57014

Taskid: 2326514
Related: odoo/upgrade#1739
Signed-off-by: Yannick Tivisse (yti) <yti@odoo.com>

================================= pseudo patch: =================================

--- a/addons/pad/models/__init__.py
+++ b/addons/pad/models/__init__.py
@@ -3,4 +3,3 @@
 
 from . import res_config_settings
 from . import pad
-from . import res_company

--- a/addons/pad/models/pad.py
+++ b/addons/pad/models/pad.py
@@ -22,15 +22,13 @@ class PadCommon(models.AbstractModel):
 
     @api.model
     def pad_is_configured(self):
-        return bool(self.env.company.pad_server)
+        return bool(self.env['ir.config_parameter'].sudo().get_param('pad.pad_server'))
 
     @api.model
     def pad_generate_url(self):
-        company = self.env.user.sudo().company_id
-
         pad = {
-            "server": company.pad_server,
-            "key": company.pad_key,
+            "server": self.env['ir.config_parameter'].sudo().get_param('pad.pad_server'),
+            "key": self.env['ir.config_parameter'].sudo().get_param('pad.pad_key'),
         }
 
         # make sure pad server in the form of http://hostname
@@ -75,8 +73,11 @@ class PadCommon(models.AbstractModel):
 
     @api.model
     def pad_get_content(self, url):
-        company = self.env.user.sudo().company_id
-        myPad = EtherpadLiteClient(company.pad_key, (company.pad_server or '') + '/api')
+        pad = {
+            "server": self.env['ir.config_parameter'].sudo().get_param('pad.pad_server'),
+            "key": self.env['ir.config_parameter'].sudo().get_param('pad.pad_key'),
+        }
+        myPad = EtherpadLiteClient(pad['key'], (pad['server'] or '') + '/api')
         content = ''
         if url:
             split_url = url.split('/p/')
@@ -131,8 +132,11 @@ class PadCommon(models.AbstractModel):
         # Update the pad if the `pad_content_field` is modified
         for k, field in self._fields.items():
             if hasattr(field, 'pad_content_field') and vals.get(field.pad_content_field) and self[k]:
-                company = self.env.user.sudo().company_id
-                myPad = EtherpadLiteClient(company.pad_key, (company.pad_server or '') + '/api')
+                pad = {
+                    "server": self.env['ir.config_parameter'].sudo().get_param('pad.pad_server'),
+                    "key": self.env['ir.config_parameter'].sudo().get_param('pad.pad_key'),
+                }
+                myPad = EtherpadLiteClient(pad['key'], (pad['server'] or '') + '/api')
                 path = self[k].split('/p/')[1]
                 myPad.setHtmlFallbackText(path, vals[field.pad_content_field])
 

--- a/addons/pad/models/res_company.py
+++ b/None
@@ -1,11 +0,0 @@
-# -*- coding: utf-8 -*-
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import fields, models
-
-
-class ResCompany(models.Model):
-    _inherit = 'res.company'
-
-    pad_server = fields.Char(help="Etherpad lite server. Example: beta.primarypad.com")
-    pad_key = fields.Char('Pad Api Key', help="Etherpad lite api key.", groups="base.group_system")

--- a/addons/pad/models/res_config_settings.py
+++ b/addons/pad/models/res_config_settings.py
@@ -7,5 +7,5 @@ from odoo import fields, models
 class ResConfigSettings(models.TransientModel):
     _inherit = 'res.config.settings'
 
-    pad_server = fields.Char(related='company_id.pad_server', string="Pad Server", readonly=False)
-    pad_key = fields.Char(related='company_id.pad_key', string="Pad API Key", readonly=False)
+    pad_server = fields.Char(config_parameter='pad.pad_server', string="Pad Server")
+    pad_key = fields.Char(config_parameter='pad.pad_key', string="Pad API Key")
